src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ConnectionHandlerContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.IO;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.Handlers;
+
+namespace Microsoft.Data.SqlClient.Microsoft.Data.SqlClientX.Handlers
+{    /// <summary>
+     /// Class that contains all context data needed for various handlers.
+     /// </summary>
+    internal class ConnectionHandlerContext : HandlerRequest
+    {
+        /// <summary>
+        /// Class that contains data required to handle a connection request.
+        /// </summary>
+        public String connectionString { get; set; }
+
+        /// <summary>
+        /// Stream used by readers.
+        /// </summary>
+        public Stream ConnectionStream { get; set; }
+
+        /// <summary>
+        /// Class required by DataSourceParser and Transport layer.
+        /// </summary>
+        public DataSource dataSource { get; set; }
+
+        /// <summary>
+        /// Class used by orchestrator while chaining handlers.
+        /// </summary>
+        public Exception error { get; set; }
```

Comment:
The base class already has an exception being exposed isnt it?

Reply:
Should it be a SqlException? What other kinds of exceptions do we expect?

Reply:
@cheenamalhotra  
I dont think I can tell you about the Class Types of the exceptions, but definitely a lot of transport related errors, (broken connections, timeouts), timeouts, 

Then there would errors coming from MSAL for AccessToken retrieval.

There is always OOM. 

While parsing the protocol, we definitely respond to Error tokens with SqlException messages. 

I feel we will need to stay away from SqlException in the base class since SqlException is only supposed to have the exceptions that is related to Sql Connectivity/operations etc. 

So likely we need some kind of indicator whether the Exception in the HandlerContext should be considered a SqlException or not. 

e.g. for any errors with driver implementation because of which we couldn't read a byte, we throw a generic exception 

![image](https://github.com/dotnet/SqlClient/assets/1623701/c8126703-ade4-4013-8bfa-274bf83c2fa9)

Reply:
Also when we try to execute a command on the same connection, twice, I believe we get an `InvalidOperation` exception.

Reply:
Agreed (I think) - I think the type of the exception in this class should stay as `Exception`

Reply:
We can use SqlErrorCollection to add multiple errors to the SqlException